### PR TITLE
Expound README

### PR DESCRIPTION
**Pull request by ari-becker**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,142 @@
...
+Kubernetes cluster operation. This is unacceptable in production clusters.
+
+Proposed solutions that attempt to solve this problem with a
+`PodDisruptionBudget` (like Elastic's own [cloud-on-k8s][cloud-on-k8s] project)
+are naive and insufficient. Elastic's official approach is to not report a
+given Elasticsearch pod as ready until [the entire cluster is green][es-cluster-health].
+However, if the cluster is momentarily yellow, then this results in the entire
+cluster becoming unavailable, with cascading failures in dependent services
+which are still functional, despite the cluster being in a yellow (i.e.
+underr-replicated, but not unavailable) state. The more mature approach is to
```
Comment (oded-dd) [inline README.md]: underr --> under

File: README.md [hunk 2]
```diff
@@ -1,20 +1,142 @@
...
+as a readiness check, but this no longer couples Kubernetes's understanding of
+readiness to Elasticsearch's notion of shard availability. Therefore, the fact
+that specific Pods in Kubernetes are available or unavailable, and that
+specific PodDisruptionBudgets are satisfied or unsatisfied, is no longer in
+and of itself sufficient to safely signal to cluster tooling whether it is safe
+to terminate the underlying Kubernetes nodes.
+
+[cloud-on-k8s][https://github.com/elastic/cloud-on-k8s]
+[es-cluster-health](https://github.com/elastic/helm-charts/blob/ffd109085023a37211c259302e2d076d84eeca94/elasticsearch/values.yaml#L228)
+
```
Comment (oded-dd) [inline README.md:57]: Use case 2 - streaming ... Kafka mechanism + lag (translated to business SLO/SLI)

Comment (oded-dd) [reply README.md:57]: Use case 3 - disruptive behavior - TME/SalesEngineer can run an important demo and block the upgrade process ...

**Review by oded-dd — CHANGES_REQUESTED** [review]
Looks much much better. 
I would also add an example directory. with the Unready alert for each of the use cases. 
It will provide the entire loop for any developer/ops who wants to use it

File: README.md (hunk 1 above)
Comment (ari-becker) [reply README.md]: fixed already

**Review by oded-dd — APPROVED** [review]

**Pull request merged**